ext/dom/php_dom.stub.php

```diff
@@ -1036,119 +1109,503 @@ function dom_import_simplexml(object $node): DOMElement {}
+        /** @readonly */
+        public ?Element $parentElement;
+        /** @implementation-alias DOMNode::hasChildNodes */
+        public function hasChildNodes(): bool {}
+        /** @readonly */
+        public NodeList $childNodes;
+        /** @readonly */
+        public ?Node $firstChild;
+        /** @readonly */
+        public ?Node $lastChild;
+        /** @readonly */
+        public ?Node $previousSibling;
+        /** @readonly */
+        public ?Node $nextSibling;
 
-        public ?string $documentURI;
+        public ?string $nodeValue;
+        public ?string $textContent;
+        /** @implementation-alias DOMNode::normalize */
+        public function normalize(): void {}
+
+        /** @implementation-alias DOMNode::cloneNode */
+        public function cloneNode(bool $deep = false): Node {}
+        public function isEqualNode(?Node $otherNode): bool {}
+        public function isSameNode(?Node $otherNode): bool {}
+
+        public const int DOCUMENT_POSITION_DISCONNECTED = 0x01;
+        public const int DOCUMENT_POSITION_PRECEDING = 0x02;
+        public const int DOCUMENT_POSITION_FOLLOWING = 0x04;
+        public const int DOCUMENT_POSITION_CONTAINS = 0x08;
+        public const int DOCUMENT_POSITION_CONTAINED_BY = 0x10;
+        public const int DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC = 0x20;
+        public function compareDocumentPosition(Node $other): int {}
+        public function contains(?Node $other): bool {}
+
+        public function lookupPrefix(?string $namespace): ?string {}
+        /** @implementation-alias DOMNode::lookupNamespaceURI */
+        public function lookupNamespaceURI(?string $prefix): ?string {}
+        public function isDefaultNamespace(?string $namespace): bool {}
+
+        public function insertBefore(Node $node, ?Node $child): Node {}
+        public function appendChild(Node $node): Node {}
+        public function replaceChild(Node $node, Node $child): Node {}
+        public function removeChild(Node $child): Node {}
+
+        /** @implementation-alias DOMNode::getLineNo */
+        public function getLineNo(): int {}
+        public function getNodePath(): string {}
+
+        /** @implementation-alias DOMNode::C14N */
+        public function C14N(bool $exclusive = false, bool $withComments = false, ?array $xpath = null, ?array $nsPrefixes = null): string|false {}
+        /** @implementation-alias DOMNode::C14NFile */
+        public function C14NFile(string $uri, bool $exclusive = false, bool $withComments = false, ?array $xpath = null, ?array $nsPrefixes = null): int|false {}
+
+        /** @implementation-alias DOMNode::__sleep */
+        public function __sleep(): array {}
+        /** @implementation-alias DOMNode::__wakeup */
+        public function __wakeup(): void {}
+    }
+
+    class NodeList implements IteratorAggregate, Countable
+    {
+        /** @readonly */
+        public int $length;
+
+        /**
+         * @tentative-return-type
```

Comment:
Why tentative? It's a new class?

Reply:
This is to implement the `count()` method from `Countable`, which is defined as tentative:
```php
interface Countable
{
    /** @tentative-return-type */
    public function count(): int;
}
```

Reply:
But surely one can implement an interface and not have tentative?